cg/store/filters/status_case_filters.py

```diff
@@ -200,6 +200,30 @@ def filter_compressible_cases(cases: Query, **kwargs) -> Query:
     return cases.filter(Case.is_compressible)
 
 
+def get_not_received_cases(cases: Query, **kwargs) -> Query:
+    return cases.filter(Sample.received_at == None).distinct()
+
+
+def get_received_cases(cases: Query, **kwargs) -> Query:
+    return cases.filter(Sample.received_at != None).distinct()
+
+
+def get_not_prepared_cases(cases: Query, **kwargs) -> Query:
+    return cases.filter(Sample.prepared_at == None).distinct()
+
+
+def get_prepared_cases(cases: Query, **kwargs) -> Query:
+    return cases.filter(Sample.prepared_at != None).distinct()
+
+
+def get_not_sequenced_cases(cases: Query, **kwargs) -> Query:
+    return cases.filter(Sample.last_sequenced_at == None).distinct()
```

Comment:
All of the filters above rely on the `cases` query being passed having been joined with the samples table. Do we place these filters elsewhere usually?

I could also do the join in the filters, but I think that would misbehave when chaining them.

Reply:
I would move the filters on sample to that file probably.

Reply:
Yeah, I do use that function and pass it :+1: Do you think these functions belong in that same module (`cg/store/base.py`)?

Reply:
No, not `base.py` but I would probably let them be in the `store/filters/status_case_sample_filters.py` file.

Reply:
I think we have something like this somehwere already.

We would start with a join query on both case and sample and then apply case and sample filters for whatever we need. 

Example:
```Python
    def get_not_analysed_cases_by_sample_internal_id(
        self,
        sample_internal_id: str,
    ) -> list[Case]:
        """Get not analysed cases by sample internal id."""

        query: Query = self._get_join_case_and_sample_query()

        not_analysed_cases: Query = apply_case_filter(
            cases=query,
            filter_functions=[
                CaseFilter.NOT_ANALYSED,
            ],
        )

        return apply_sample_filter(
            samples=not_analysed_cases,
            filter_functions=[SampleFilter.BY_INTERNAL_ID],
            internal_id=sample_internal_id,
        ).all()
```

Reply:
Although I think this can be done with a subquery within the filter as well. It would make the filter more complex, guess it depends on where we want to go with all of this.

Reply:
Ah, that query is perfect :+1: Thanks, I think I'll stick to that syntax for now even if I like the subqueries. I feel like the subquery syntax is a little more expressive, but I'll stick to what I know for now

Reply:
Moved the filters in https://github.com/Clinical-Genomics/cg/pull/3050/commits/4ac0d95aa18f26666310f54174ef24e7f82bfbb6